> # Proyecto Entrega 1
> ##### &nbsp;&nbsp;&nbsp;&nbsp; Análisis Exploratorio Univariado
> - **Fecha**: 18/09/2023  |  ITESO  
> - **Materia**: Programación para el Análisis de Datos
> - **Alumnos**: 
>    - Gabriel Alejandro Olvera González is734604
>    - Michelle Figueroa Jacques is727328
>    - Miguel Mario Méndez Alcaráz is2
>    - Sofía Vargas Aceves is727375  
>  <br>

## Descripción del DataSet
Este Dataset es proporcionado por el CONEVAL, e incluye datos de la pobreza en México 2022. 
El dataset utiliza como fuente de información la Encuesta Nacional de Ingresos y Gastos de los Hogares (ENIGH) 2022, publicada por el Instituto Nacional de Estadística y Geografía (INEGI) el 26 de julio de 2023.

Es el resultado de entrevistas a familias mexicanas de bajos recursos con el propósito de analizar los factores determinantes de la pobreza en México. Contiene una amplia gama de variables que abarcan desde identificadores de viviendas, hogares y personas, hasta información demográfica, socioeconómica y de calidad de vida. Estas variables incluyen datos sobre educación, acceso a servicios de salud y seguridad social, condiciones de vivienda, ingresos, programas sociales, población vulnerable y discapacitada, entre otros. Este conjunto de datos proporciona una visión completa de las condiciones de vida de estas familias y es valioso para la formulación de políticas públicas y programas de desarrollo destinados a combatir la pobreza en México.​

+ Columnas: Cuenta con 79 columnas de información del encuestado e índices calculados por el CONEVAL.
+ Filas: Hay 309,534, indicando que este es el **tamaño de la muestra**
+ Los datos van de acuerdo con el descriptor de la base de la ENIGH 2022 del INEGI.

#### Tipo de datos
> + Estos datos son **transversales**. Esto se debe a que no se recopilan datos a través del tiempo. Estos datos son únicamente del 2022.
> + Estos datos **no son de panel**. Dado que sólo se toman los datos de un encuestado sólo una vez y no se toman los datos de éste a lo largo del tiempo.

#### Unidad de observación
Cada fila de la tabla...  
>  <br>  
>  Encuestado dentro de familia de bajos recursos en México 2022
>  <br> 
>   <br>      

#### Fuentes
> +  CONEVAL (2022). Medición de Pobreza Multidimensional Mexico 2022​. Recuperado de: https://www.kaggle.com/datasets/davidmaldonadoc/2022-mexicos-poverty-survey?resource=download ​
> + Maldonado, D. (2022). 2022 Mexico's poverty survey. Kaggle. Recuperado de: https://www.kaggle.com/datasets/davidmaldonadoc/2022-mexicos-poverty-survey?resource=download&select=Pobreza+2022+CONEVAL.py


## Variables

### Llaves

| __ID__ | __Rango__ | __Tipo__ | __Descripción__ |
| --- | --- | --- | --- | 
| folioviv  | {100005002,3260797912} | nominal   | Identificador de la vivienda
| foliohog  | {1, 5}                 | nominal   | Identificador del hogar
| numren    | {1, 19}                | ordinal   | Identificador de la persona

### De estadística y localización

| __ID__ | __Rango__ | __Tipo__ | __Descripción__ | __Categorías__
| --- | --- | --- | --- | --- |
| ent       | {1, 32}       | nominal   | Identificador de la entidad federativa
| est_dis   | {1, 560}      | nominal   |  Estrato de diseño 
| factor    | {6, 6470}     |           | Factor de expansión
| rururb    | {0,1}         | nominal   | Identificador de localidades rurales      | 0)Urbana 1)Rural
| tam_loc   | {1, 4}        | ordinal   | Tamaño de la localidad                    | 1)>= 100k ,  2)[15k,100k) ,  3)[2.5k,15k) ,  4)<2.5k
| ubica_geo | {1001, 32057} |           | Ubicación geográfica
| upm       | {1, 10232}    | nominal   | Unidad primaria de muestreo    

### Información del encuestado

#### Información personal


| __ID__ | __Rango__ | __Tipo__ | __Descripción__ | __Categorías__
| --- | --- | --- | --- | --- |
| a_nacim   | {1913,2022}   | razón     | Año de nacimiento
| discap    | {0,1}         | nominal   | Población con presencia de discapacidad física o mental | 0)Sin 1)Con
| edad      | {0, 109}      | intervalo | Edad reportada al momento de la entrevista
| jub       | {0,1}         | nominal   | Población pensionada o jubilada           | 0)No 1)Sí
| hli       | {0,1}         | nominal   | Hablante de lengua indígena               | 0)No 1)Sí
| ed_inas  | {0,1}         | nominal   | Inasistencia a la escuela                 | 0)Sí asiste 1)No asiste
| ed_niv | {0,3} | ordinal | Nivel educativo | 0)<Primaria , 1)[Primaria,Secundaria) , 2)[Secundaria,Media superior) , 3)[Media superior,Mayor nivel educativo]
| p_menor   | {0,1}         | nominal   | Hogares con población de 0 a 17 años      | 0)Sin menores 1)Con menores
| pam       | {0,1}         | nominal   | Programa de adultos mayores               | 0)No recibe 1)Sí recibe
| sexo      | {1,2}         | nominal   | Sexo                                      | 1)Masculino 2)Femenino
| tamhogesc | {1, 16.60}    | razón     | Tamaño del hogar ajustado                 
| parentesco| {101,623}     | TODO      | Parentesco con el jefe del hogar (INEGI)


#### Información de salud y seguro


| __ID__ | __Rango__ | __Tipo__ | __Descripción__ | __Categorías__
| --- | --- | --- | --- | --- |
| ali_ins     | {0,3} | ordinal | Grado de Inseguridad Alimentaria | 0)Seguridad alimentaria 1)Inseguridad leve 2)Inseguridad moderada 3)Inseguridad severa
| ali_lim     | {0,1}         | nominal   | Limitación en el consumo de alimentos                      | 0)No limitado 1)Limitado
| ali_dieta   | {1,3}         | ordinal   | Dieta consumida en los hogares                             | 1)Pobre 2)Limítrofe 3) Aceptable
| salud_s     | {0,1}         | nominal   | Servicios médicos por otros núcleos familiares o por contratación propia | 0)No cuenta 1)Sí cuenta
| salud_s_dir | {0,1}         | nominal   | Acceso directo a servicios de salud       | 0)Sin acceso 1)Con acceso
| ss_dir      | {0,1}         | nominal   | Acceso directo a seguridad social         | 0)Sin acceso 1)Con acceso
| ss_jef      | {0,1}         | nominal   | Acceso directo a seguridad social de jefatura del hogar| 0)No cuenta 1)Sí cuenta
| ss_cony     | {0,1}         | nominal   | Acceso directo a seguridad social de cónyugue de jefatura del hogar| 0)No cuenta 1)Sí cuenta
| ss_hijo     | {0,1}         | nominal   | Acceso directo a seguridad social de hijo de jefatura del hogar| 0)No cuenta 1)Sí cuenta
| par         | {1,6}         | nominal   | Integrantes que tienen acceso por otros miembros  | TODO ... acceso a qué?


#### Información económica


| __ID__ | __Rango__ | __Tipo__ | __Descripción__ | __Categorías__
| --- | --- | --- | --- | --- |
| ing_t     | {0, 2444444.2} | intervalo | Ingreso corriente total del hogar 
| ing_t_pc  | {0, 1057393.8} | intervalo | Ingreso corriente total per cápita
| ing_mon   | {0, 2444444.2} | intervalo | Ingreso corriente monetario del hogar
| ing_lab   | {0, 2444444.2} | intervalo | Ingreso corriente monetario laboral
| ing_ren   | {0, 252623.5}  | intervalo | Ingreso corriente monetario por rentas
| ing_tra   | {0, 238695.17} | intervalo | Ingreso corriente monetario por transferencias
| ing_nomon     | {0, 1012431.1} | intervalo | Ingreso corriente no monetario
| ing_pago_esp  | {0, 1012431.1} | intervalo | Ingreso corriente no monetario pago especie
| ing_reg_esp   | {0, 326264.29} | intervalo | Ingreso corriente no monetario regalos especie
| ing_pam       | {0, 14525.06}  | intervalo | Ingreso promedio mensual por programas de adultos mayores 
| pea           | {0,2}          | nominal   | Población económicamente activa           | 0)PNEA 1)PEA:Ocupada 2)PEA:Desocupada  


### Indicadores y Escalas

#### Indicadores de carencias


| __ID__ | __Rango__ | __Tipo__ | __Descripción__ | __Categorías__
| --- | --- | --- | --- | --- |
| ic_1        | {0, 1}        | nominal   | Población con al menos una carencia social                 | 0)Sin carencias 1)Con al menos una carencia
| ic_3        | {0, 1}        | nominal   | Población con al menos tres carencias sociales             | 0)Sin carencias 1)Con al menos 3 carencias
| ic_ali      | {0,1}         | nominal   | Por acceso a la alimentación         | 0)No presenta carencia 1)Presenta carencia
| ic_ali_nc   | {0,1}         | nominal | Por acceso a alimentación nutritiva de calidad | 0)No presenta carencia 1)Presenta carencia
| ic_salud_s  | {0,1}         | nominal   | Por acceso a servicios de salud      | 0)No presenta carencia 1)Presenta carencia
| ic_rezedu | {0,1}         | nominal   | Indicador de rezago educativo                              | 0)No hay rezago        1)Presenta rezago
| ic_segsoc | {0,1}         | nominal   | Por acceso a seguridad social        | 0)No presenta carencia 1)Presenta carencia
| icv       | {0,1}         | nominal   | Por calidad y espacios de vivienda   | 0)No presenta carencia 1)Presenta carencia
| icv_pisos | {0,1}         | nominal   | Por material de piso de vivienda     | 0)No presenta carencia 1)Presenta carencia
| icv_muros | {0,1}         | nominal   | Por material de muros de vivienda    | 0)No presenta carencia 1)Presenta carencia
| icv_techos| {0,1}         | nominal   | Por material de techos de vivienda   | 0)No presenta carencia 1)Presenta carencia
| icv_hac   | {0,1}         | nominal   | Por hacinamiento de piso de vivienda | 0)No presenta carencia 1)Presenta carencia
| icsb      | {0,1}         | nominal   | Por acceso a servivios básicos de vivienda | 0)No presenta carencia 1)Presenta carencia
| icsb_agua | {0,1}         | nominal   | Por acceso al agua                   | 0)No presenta carencia 1)Presenta carencia
| icsb_dren | {0,1}         | nominal   | Por servicio de drenaje              | 0)No presenta carencia 1)Presenta carencia
| icsb_luz  | {0,1}         | nominal   | Por servicios de electricidad        | 0)No presenta carencia 1)Presenta carencia
| icsb_comb | {0,1}         | nominal   | Por combustible para cocinar         | 0)No presenta carencia 1)Presenta carencia


#### Otros indicadores y escalas


| __ID__ | __Rango__ | __Tipo__ | __Descripción__ | __Categorías__
| --- | --- | --- | --- | --- |
| cuadrantes | {1,4}        | nominal | Cuadrantes de bienestar económico y derechos sociales | 1)Pobres 2)Vulnerabes por carencia 3)Vulnerables por ingresos 4)No pobres y no vulnerables
| ia_adulto     | {0, 6}    |         | Escala de IA para hogares sin menores de 18 años
| ia_menor      | {0, 12}   |         | Escala de IA para hogares con menores de 18 años
| int_pob       | {0, 1}    |         | Intensidad de la privación social: en situación de pobreza
| int_pob_e     | {0, 1}    |         | Intensidad de la privación social: en situación de pobreza extrema
| int_vul_ic    | {0, 1}    |         | Intensidad de la privación social: población vulnerable por carencias
| int_ic        | {0, 1}    |         | Intensidad de la privación social: población vulnerable por carencias sociales    
| pobreza       | {0,1}     | nominal | Pobreza                                                                     | 0)No 1) Sí
| pobreza_l_ing  | {0,1}     | nominal | Población con ingreso menor a la línea de pobreza extrema por ingresos      | 0)>=línea 1)<línea
| pobreza_prof   | {0, 1}    |         | Índice FGT con alfa igual a 1 (línea de pobreza por ingresos)
| pobreza_e       | {0,1}     | nominal   | Pobreza extrema                                                             | 0)No 1) Sí
| pobreza_e_l_ing  | {0,1}     | nominal   | Población con ingreso menor a la línea de pobreza extrema por ingresos      | 0)>=línea 1)<línea
| pobreza_e_prof | {0, 1}    |           | Índice FGT con alfa igual a 1 (línea de pobreza extrema por ingresos)
| pobreza_m     | {0,1}     | nominal   | Pobreza moderada                         | 0)No 1) Sí
| priv          | {0, 6}    |           | Índice de Privación Social
| priv_prof     | {0, 1}    | nominal   | Profundidad de la privación social
| vul_ing       | {0,1}     | nominal   | Población vulnerable por ingresos        | 0)No 1)Sí
| vul_ic        | {0,1}     | nominal   | Población vulnerable por carencias       | 0)No 1)Sí
| no_pob_vul    | {0,1}     | nominal   | Población no pobre y no vulnerable       | 0)Pobre y vulnerable 1)No pobre y no vulnerable


### Preguntas de investigación

1. ¿Cuáles son los principales determinantes de la pobreza extrema por ingresos en México, considerando variables como educación, acceso a servicios de salud y seguridad social, y calidad de vivienda, y cómo varían estos factores entre diferentes entidades federativas?​

2. ¿Existe una correlación significativa entre el nivel de ingresos per cápita en los hogares y la prevalencia de inseguridad alimentaria severa, y qué medidas de intervención podrían reducir esta inseguridad en las poblaciones más vulnerables?​

3. ¿Cuál es el impacto del acceso a programas de apoyo social, como el Programa de Adultos Mayores, en la reducción de la pobreza y la mejora de la calidad de vida de las personas mayores en hogares de bajos recursos?​

4. ¿Cómo se relacionan los indicadores de privación social y las carencias sociales con la intensidad de la privación social, y qué estrategias podrían implementarse para abordar de manera efectiva estas carencias y mejorar la calidad de vida en hogares vulnerables?​

5. ¿Cuál es el efecto de la lengua indígena y la discapacidad en el acceso a servicios de salud, educación y empleo, y cómo se pueden diseñar políticas inclusivas para mejorar la participación y el bienestar de estas poblaciones en riesgo en diferentes regiones de México?


#### Unidad de análisis

| __Pregunta de investigación__ | __Unidad de análisis__ |
| --------------------------------------------------- | ------------------ | 
| ¿Cuáles son los principales determinantes de la pobreza extrema por ingresos en México y cómo varían estos factores entre diferentes entidades federativas?​ | entidad federativa |  
| ---------------------------------------------------- | ------------------ | 
| ¿Existe una correlación significativa entre el nivel de ingresos per cápita en los hogares y la prevalencia de inseguridad alimentaria severa, y qué medidas de intervención podrían reducir esta inseguridad en las poblaciones más vulnerables?​ | encuestados con inseguridad alimentaria y vulnerabilidad |
| ---------------------------------------------------- | ------------------ | 
| ¿Cuál es el impacto del acceso a programas de apoyo social, como el Programa de Adultos Mayores, en la reducción de la pobreza y la mejora de la calidad de vida de las personas mayores en hogares de bajos recursos?​       | adultos mayores y adultos mayores en el PAM |
| ---------------------------------------------------- | ------------------ | 
| ¿Cómo se relacionan los indicadores de privación social y las carencias sociales con la intensidad de la privación social, y qué estrategias podrían implementarse para abordar de manera efectiva estas carencias y mejorar la calidad de vida en hogares vulnerables?​    | indicadores de privación y carencia social e intensidad de privación |
| ---------------------------------------------------- | ------------------ | 
| ¿Cuál es el efecto de la lengua indígena y la discapacidad en el acceso a servicios de salud, educación y empleo, y cómo se pueden diseñar políticas inclusivas para mejorar la participación y el bienestar de estas poblaciones en riesgo en diferentes regiones de México? | hablantes de lenguas indígenas y con discapacidad |


## Procesamiento del Dataset

### Librerías

#### Análisis

In [366]:
import numpy as np
import pandas as pd

#### Salidas y visualización

In [367]:
# Para obtener múltiples salidas de una celda
from IPython.core.interactiveshell import InteractiveShell
from IPython.display import display
InteractiveShell.ast_node_interactivity = "all"

In [368]:
import matplotlib.pyplot as plt
import seaborn as sns

In [369]:
#from ipywidgets import interact
#%matplotlib inline

### Variables

#### Lectura del Dataset  
Primero se hace una lectura del dataset, junto con una copia de éste.

In [370]:
df = pd.read_csv('../pobreza22.csv',encoding='utf-8')

In [371]:
df_backup = df.copy()
df

folioviv  foliohog  numren  est_dis    upm  factor  tam_loc  rururb  \
0        100005002         1       1        3      1     206        1     0.0   
1        100005002         1       2        3      1     206        1     0.0   
2        100005002         1       3        3      1     206        1     0.0   
3        100005003         1       1        3      1     206        1     0.0   
4        100005003         1       2        3      1     206        1     0.0   
...            ...       ...     ...      ...    ...     ...      ...     ...   
309529  3260797910         1       1      560  10232     123        4     1.0   
309530  3260797910         1       2      560  10232     123        4     1.0   
309531  3260797912         1       1      560  10232     123        4     1.0   
309532  3260797912         1       2      560  10232     123        4     1.0   
309533  3260797912         1       3      560  10232     123        4     1.0   

        ent  ubica_geo  ...           ict       ing_mon       ing_lab  \
0         1       1001  ...  15282.204133  15282.204133  12254.222301   
1         1       1001  ...  15282.204133  15282.204133  12254.222301   
2         1       1001  ...  15282.204133  15282.204133  12254.222301   
3         1       1001  ...  36889.050990  33950.029778  22901.587447   
4         1       1001  ...  36889.050990  33950.029778  22901.587447   
...     ...        ...  ...           ...           ...           ...   
309529   32      32046  ...  51414.758596  46788.301738  46788.301738   
309530   32      32046  ...  51414.758596  46788.301738  46788.301738   
309531   32      32046  ...  14101.618920  13259.491956  13259.491956   
309532   32      32046  ...  14101.618920  13259.491956  13259.491956   
309533   32      32046  ...  14101.618920  13259.491956  13259.491956   

             ing_ren      ing_tra        nomon  pago_esp      reg_esp  hli  \
0           0.000000  3027.981832     0.000000       0.0     0.000000  0.0   
1           0.000000  3027.981832     0.000000       0.0     0.000000  0.0   
2           0.000000  3027.981832     0.000000       0.0     0.000000  0.0   
3       11048.442332     0.000000  2939.021211       0.0  2939.021211  0.0   
4       11048.442332     0.000000  2939.021211       0.0  2939.021211  0.0   
...              ...          ...          ...       ...          ...  ...   
309529      0.000000     0.000000  4626.456858       0.0  4626.456858  0.0   
309530      0.000000     0.000000  4626.456858       0.0  4626.456858  0.0   
309531      0.000000     0.000000   842.126965       0.0   842.126965  0.0   
309532      0.000000     0.000000   842.126965       0.0   842.126965  0.0   
309533      0.000000     0.000000   842.126965       0.0   842.126965  NaN   

        discap  
0          0.0  
1          0.0  
2          0.0  
3          0.0  
4          0.0  
...        ...  
309529     0.0  
309530     0.0  
309531     0.0  
309532     0.0  
309533     0.0  

[309534 rows x 79 columns]

#### Renombramientos de Columnas


In [372]:
# Lista de nombres de columnas
col = df.columns
col

Index(['folioviv', 'foliohog', 'numren', 'est_dis', 'upm', 'factor', 'tam_loc',
       'rururb', 'ent', 'ubica_geo', 'edad', 'sexo', 'parentesco', 'ic_rezedu',
       'anac_e', 'inas_esc', 'niv_ed', 'ic_asalud', 'ic_segsoc', 'sa_dir',
       'ss_dir', 's_salud', 'par', 'jef_ss', 'cony_ss', 'hijo_ss', 'pea',
       'jub', 'pam', 'ing_pam', 'ic_cv', 'icv_pisos', 'icv_muros',
       'icv_techos', 'icv_hac', 'ic_sbv', 'isb_agua', 'isb_dren', 'isb_luz',
       'isb_combus', 'ic_ali_nc', 'id_men', 'tot_iaad', 'tot_iamen', 'ins_ali',
       'ic_ali', 'lca', 'dch', 'plp_e', 'plp', 'pobreza', 'pobreza_e',
       'pobreza_m', 'vul_car', 'vul_ing', 'no_pobv', 'i_privacion',
       'carencias', 'carencias3', 'cuadrantes', 'prof1', 'prof_e1', 'profun',
       'int_pob', 'int_pobe', 'int_vulcar', 'int_caren', 'tamhogesc', 'ictpc',
       'ict', 'ing_mon', 'ing_lab', 'ing_ren', 'ing_tra', 'nomon', 'pago_esp',
       'reg_esp', 'hli', 'discap'],
      dtype='object')

In [373]:
# Llaves
    # Permanecen iguales ['folioviv','foliohog','numren']
# Estadística
    # Permanecen iguales ['ent','est_dis','factor','rururb','tam_loc','ubica_geo','upm']
# Información personal
    # Permanecen iguales:  ['discap','edad','jub','hli','pam','sexo','tamhogesc','parentesco']
df = df.rename(
    columns= {'anac_e':'a_nacim', 
              'inas_esc':'ed_inas',
              'niv_ed':'ed_niv',
              'id_men':'p_menor',
              }
)
# Información de salud y seguro
        # Permanecen iguales:  ['ss_dir','par']
df = df.rename(
    columns= {'ins_ali':'ali_ins',
              'lca':'ali_lim',
              'dch':'ali_dieta',
              's_salud':'salud_s',
              'sa_dir':'salud_s_dir',
              'jef_ss':'ss_jef',
              'cony_ss':'ss_cony',
              'hijo_ss':'ss_hijo',
              }
)
# Información económica
        # Permanecen iguales:  ['ing_mon','ing_lab','ing_ren','ing_tra','ing_pam','pea',]
df = df.rename(
    columns= {'ict':'ing_t',
              'ictpc':'ing_t_pc',
              'nomon':'ing_nomon',
              'pago_esp':'ing_pago_esp',
              'reg_esp':'ing_reg_esp',
              }
)
# Indicadores de carencias
        # Permanecen iguales:  ['ic_ali','ic_ali_nc','ic_rezedu','ic_segsoc','icv_pisos','icv_muros','icv_techos','icv_hac',]
df = df.rename(
    columns= {'carencias':'ic_1',
              'carencias3':'ic_3', 
              'ic_asalud':'ic_salud_s',
              'ic_cv':'icv',
              'ic_sbv':'icsb',
              'isb_agua':'icsb_agua',
              'isb_dren':'icsb_dren',
              'isb_luz':'icsb_luz',
              'isb_combus':'icsb_comb',
              }
)
# Otros indicadores y escalas
        # Permanecen iguales:  ['cuadrantes','int_pob','pobreza', 'pobreza_e','pobreza_m','vul_ing',]
df = df.rename(
    columns= {'tot_iaad':'ia_adulto',
              'tot_iamen':'ia_menor',
              'int_pobe':'int_pob_e',
              'int_vulcar':'int_vul_ic',
              'int_caren':'int_ic', 
              'plp':'pobreza_l_ing',
              'prof1': 'pobreza_prof',
              'plp_e':'pobreza_e_l_ing',
              'prof_e1': 'pobreza_e_prof',
              'i_privacion':'priv',
              'profun':'priv_prof',
              'vul_car':'vul_ic',
              'no_pobv':'no_pob_vul',

              }
)

In [374]:
col = df.columns
col

Index(['folioviv', 'foliohog', 'numren', 'est_dis', 'upm', 'factor', 'tam_loc',
       'rururb', 'ent', 'ubica_geo', 'edad', 'sexo', 'parentesco', 'ic_rezedu',
       'a_nacim', 'ed_inas', 'ed_niv', 'ic_salud_s', 'ic_segsoc',
       'salud_s_dir', 'ss_dir', 'salud_s', 'par', 'ss_jef', 'ss_cony',
       'ss_hijo', 'pea', 'jub', 'pam', 'ing_pam', 'icv', 'icv_pisos',
       'icv_muros', 'icv_techos', 'icv_hac', 'icsb', 'icsb_agua', 'icsb_dren',
       'icsb_luz', 'icsb_comb', 'ic_ali_nc', 'p_menor', 'ia_adulto',
       'ia_menor', 'ali_ins', 'ic_ali', 'ali_lim', 'ali_dieta',
       'pobreza_e_l_ing', 'pobreza_l_ing', 'pobreza', 'pobreza_e', 'pobreza_m',
       'vul_ic', 'vul_ing', 'no_pob_vul', 'priv', 'ic_1', 'ic_3', 'cuadrantes',
       'pobreza_prof', 'pobreza_e_prof', 'priv_prof', 'int_pob', 'int_pob_e',
       'int_vul_ic', 'int_ic', 'tamhogesc', 'ing_t_pc', 'ing_t', 'ing_mon',
       'ing_lab', 'ing_ren', 'ing_tra', 'ing_nomon', 'ing_pago_esp',
       'ing_reg_esp', 'hli', 'discap']

## Limpieza de Datos

In [375]:
# No todas las columnas tienen valores completos
null_data = df.isnull().sum()
null_data = null_data[null_data > 0].sort_values(ascending=False)
null_data

pam           279571
ing_pam       274766
ia_adulto     217154
ia_menor       92380
pea            81462
ed_inas        11470
hli            11470
ed_niv         11470
discap           400
ic_1              14
ic_3              14
int_ic            14
int_vul_ic        14
int_pob_e         14
int_pob           14
priv_prof         14
cuadrantes        14
vul_ic            14
priv              14
no_pob_vul        14
vul_ing           14
pobreza_m         14
pobreza_e         14
pobreza           14
icsb               7
icv_pisos          7
icv                7
icsb_comb          7
dtype: int64

### Limpieza de columnas pam + ing_pam

La variable con mayor número de datos faltantes es PAM (Programa de Adultos Mayores). Según la CONEVAL, en el 2022, el programa sólo se le aplica a adultos mayores de 65 o más años de edad. De la misma manera, ing_pam sólo puede tener un valor, si el adulto mayor está dentro de programa.
* Una de las primeras acciones que podemos tomar, es poner en 0 tanto ing_pam como pam, cuando la edad del encuestado es de menor de 65 años.  

Con este cambio, a continuación se observa que ya no hay datos nulos dentro de la columna de pam, pero todavía quedan algunos datos de ing_pam con nulos.

In [376]:
# Si la edad es menor de 65, pam e ing_pam serán igual a 0
df.loc[df['edad'] < 65, ['pam', 'ing_pam']] = 0,0
# PAM
df.isnull().sum()['pam']
df[df['pam'].isna()][["edad","jub","pam","ing_pam"]]

0

Empty DataFrame
Columns: [edad, jub, pam, ing_pam]
Index: []

In [377]:
# ING PAM
df.isnull().sum()['ing_pam']
df[df['ing_pam'].isna()][["edad","jub","pam","ing_pam"]]

4222

edad  jub  pam  ing_pam
3         68  0.0  0.0      NaN
25        69  0.0  0.0      NaN
49        66  0.0  0.0      NaN
100       67  0.0  0.0      NaN
108       66  0.0  0.0      NaN
...      ...  ...  ...      ...
309368    72  0.0  0.0      NaN
309369    87  0.0  0.0      NaN
309370    96  0.0  0.0      NaN
309371    69  0.0  0.0      NaN
309419    81  0.0  0.0      NaN

[4222 rows x 4 columns]

* Para resolverlo, ahora simplemente tenemos que determinar el valor de ingresos del programa dependiendo de si el encuestado está en el programa, dado que pam ya no tiene nulos.  

Este cambio logró eliminar los nulos para ing_pam.

In [378]:
df.loc[df['pam'] == 0, ['ing_pam']] = 0
# PAM
df.isnull().sum()['ing_pam']
df[df['ing_pam'].isna()][["edad","jub","pam","ing_pam"]]

0

Empty DataFrame
Columns: [edad, jub, pam, ing_pam]
Index: []

### Limpieza de columnas ia_adulto + ia_menor

Estos valores son escalas de inseguridad alimentaria, y a partir del archivo de python en kaggle y por su definición, sabemos que todo ia_adulto corresponde a p_menor (población menor) en 0 y ia_menor a p_menor en 1. Esto es, casa sin menores y casa con menores respectivamente.  
+ Corroboramos con el dataset, viendo si la cantidad de casas con menores y casas sin menores son los causantes.

> Fuente: 
Maldonado, D. (2022). 2022 Mexico's poverty survey. Kaggle. Recuperado de: https://www.kaggle.com/datasets/davidmaldonadoc/2022-mexicos-poverty-survey?resource=download&select=Pobreza+2022+CONEVAL.py

In [379]:
print("--Dataset--")
df['p_menor'].value_counts(dropna=False)

df_limpieza_ia_a = df[df['ia_adulto'].isna()][["edad","ia_adulto","ia_menor","p_menor","ali_ins"]]
print("--IA sin menores (" + str(len(df_limpieza_ia_a)) +" valores) --")
df_limpieza_ia_a['p_menor'].value_counts(dropna=False,normalize=True)


df_limpieza_ia_m = df[df['ia_menor'].isna()][["edad","ia_adulto","ia_menor","p_menor","ali_ins"]]
print("--IA con menores--(" + str(len(df_limpieza_ia_m)) +" valores) --")
df_limpieza_ia_m['p_menor'].value_counts(dropna=False,normalize=True)

--Dataset--


1.0    217154
0.0     92380
Name: p_menor, dtype: int64

--IA sin menores (217154 valores) --


1.0    1.0
Name: p_menor, dtype: float64

--IA con menores--(92380 valores) --


0.0    1.0
Name: p_menor, dtype: float64

Podemos observar que ia_adulto y ia_menor son mutuamente excluyentes. Si p_menor es 1, se toma el valor de ia_menor. Si p_menor es 0, se toma el valor de ia_mayor.  
Por lo pronto, <ins>se pondrán en 0 los nulos, considerando que cualquier análisis de estas dos columnas, debe hacerse tomando en cuenta el valor de p_menor</ins>.

In [380]:
df.loc[df['ia_adulto'].isnull(), ['ia_adulto']] = 0
df.loc[df['ia_menor'].isnull(), ['ia_menor']] = 0

### Limpieza de columnas pea

Tenemos 81462 datos nulos en pea. A partir del archivo de python dado por DAVID MALDONADO CASTILLEJOS en Kaggle, Pobreza 2022 CONEVAL.py, podemos ver la manera en la que PEA se calcula.  Uno de los primeros puntos que llama nuestra atención, es que dice ser calculado para personas de 16 años o más.  
* Bajo ese umbral, primero observamos cuántos de los valores nulos corresponden a encuestados menores de 16 años. 

> Fuente: 
Maldonado, D. (2022). 2022 Mexico's poverty survey. Kaggle. Recuperado de: https://www.kaggle.com/datasets/davidmaldonadoc/2022-mexicos-poverty-survey?resource=download&select=Pobreza+2022+CONEVAL.py

In [381]:
df_limpieza_pea = df[df['pea'].isna()][['pea','edad']]

print('Número de encuestados con edad < 16 años: '+ str(len(df.loc[df['edad']<16])))
print('Número de nulos en pea: '+ str(df.isnull().sum()['pea']))
print('Número de encuestados de < 16 años dentro de los nulos: ' + str(len(df_limpieza_pea.loc[df_limpieza_pea['edad']<16])))

Número de encuestados con edad < 16 años: 81462
Número de nulos en pea: 81462
Número de encuestados de < 16 años dentro de los nulos: 81462


Esto quiere decir, que todos los que no tienen valores en pea, son menores de 16 años. <ins>Se decidió poner en 0 el pea de los menores a 16 años</ins>. 

In [382]:
df.loc[df['edad'] < 16, ['pea']] = 0

### Limpieza de ed_inas + ed_niv + hli

Notamos que hay un mismo número de nulos (11,470) para las columnas ed_inas, hli y ed_niv. Checamos entonces si corresponden las filas con valores faltantes para estas 3 columnas.  
A continuación, podamos corroborar que esta suposición es correcta al contar las filas que tienen valores nulos para las 3 columnas.

In [383]:
df_limpieza_ed = df[df['ed_inas'].isna()][["edad","folioviv","foliohog","numren","ed_inas","hli","ed_niv"]]    
df_limpieza_ed 
df_limpieza_ed.isnull().sum()
df_limpieza_ed.isna().sum()

edad    folioviv  foliohog  numren  ed_inas  hli  ed_niv
46         2   100023305         1       6      NaN  NaN     NaN
69         2   100050501         1      11      NaN  NaN     NaN
129        1   100071706         1       5      NaN  NaN     NaN
213        0   100134206         1       5      NaN  NaN     NaN
288        0   100158503         1       3      NaN  NaN     NaN
...      ...         ...       ...     ...      ...  ...     ...
309412     2  3260788318         1       3      NaN  NaN     NaN
309461     2  3260797213         1       3      NaN  NaN     NaN
309483     1  3260797419         1       4      NaN  NaN     NaN
309498     1  3260797423         1       5      NaN  NaN     NaN
309533     0  3260797912         1       3      NaN  NaN     NaN

[11470 rows x 7 columns]

edad            0
folioviv        0
foliohog        0
numren          0
ed_inas     11470
hli         11470
ed_niv      11470
dtype: int64

edad            0
folioviv        0
foliohog        0
numren          0
ed_inas     11470
hli         11470
ed_niv      11470
dtype: int64

Al obtener los datos en la última búsqueda, notamos que las edades eran bajas. Por lo que obtuvimos el máximo que dio un resultado de 2 años. Comparamos cómo se manejaron los valores de edades de máximo 2 años, para ed_inas, ed_niv y hli. Al hacer esta búsqueda, notamos que la misma cantidad de observaciones correspondían para todo el dataset. 
Esto significa, que <ins>todos los nulos de estas 3 columnas, son a causa de que el encuestado es un menor de máximo 3 años.</ins>  

In [384]:
df_limpieza_ed['edad'].max()
df.loc[df['edad'] < 3, ['edad','ed_niv', 'ed_inas','hli']]

2

edad  ed_niv  ed_inas  hli
46         2     NaN      NaN  NaN
69         2     NaN      NaN  NaN
129        1     NaN      NaN  NaN
213        0     NaN      NaN  NaN
288        0     NaN      NaN  NaN
...      ...     ...      ...  ...
309412     2     NaN      NaN  NaN
309461     2     NaN      NaN  NaN
309483     1     NaN      NaN  NaN
309498     1     NaN      NaN  NaN
309533     0     NaN      NaN  NaN

[11470 rows x 4 columns]

* Nuestra propuesta para solucionar esta falta de datos, es comparar cómo se manejan a los menores que tienen 3 años. 

In [385]:
df_limpieza_ed = df.loc[df['edad'] == 3, ['edad','ed_niv', 'ed_inas','hli']]

#### Nivel educativo

In [386]:
df_limpieza_ed['ed_niv'].value_counts(dropna=False,normalize=True)
df_limpieza_ed.loc[df_limpieza_ed['ed_niv']==1, ['edad','ed_niv', 'ed_inas','hli']]

0.0    0.999566
1.0    0.000434
Name: ed_niv, dtype: float64

edad  ed_niv  ed_inas  hli
8544       3     1.0      1.0  0.0
211073     3     1.0      1.0  0.0

Un 99.95% de los niños de 3 años, tienen un nivel de escolaridad menor a la primaria, en donde 0.04% (2 encuestados) tienen un nivel de educativo mayor o igual a la primaria pero menor al de secundaria. Ambos niños tienen inasistencia a la escuela, por lo que se supone que estos son casos extremos.    
Ya que hay una alta improbabilidad de que niños de 3 años tengan un nivel de primaria, y bajo la suposición de que un niño de 2 años tendrá menos probabilidad de estar en este nivel educativo, <ins>decidimos reemplazar los nulos de nivel educativo con un 0 (nivel educativo menor a la primaria)</ins>.

#### Inasistencia a la escuela

In [387]:
df_limpieza_ed['ed_inas'].value_counts(dropna=False,normalize=True)
df_limpieza_ed.loc[df_limpieza_ed['ed_inas']==1, ['edad','ed_niv', 'ed_inas','hli']]

1.0    0.558089
0.0    0.441911
Name: ed_inas, dtype: float64

edad  ed_niv  ed_inas  hli
55         3     0.0      1.0  0.0
485        3     0.0      1.0  0.0
520        3     0.0      1.0  0.0
626        3     0.0      1.0  0.0
641        3     0.0      1.0  0.0
...      ...     ...      ...  ...
308777     3     0.0      1.0  0.0
309264     3     0.0      1.0  0.0
309438     3     0.0      1.0  0.0
309449     3     0.0      1.0  0.0
309509     3     0.0      1.0  0.0

[2570 rows x 4 columns]

A partir de los valores de inasistencia a la escuela para niños de 3 años, observamos que un 55.81% no asiste, mientras que un 44.19% asiste. Considerando que sólo 2 de ellos tienen un grado de escolaridad de primaria, sabemos que aquellos que asisten a la escuela están en su mayoría, todavía debajo del nivel de primaria. Considerando que en 2022, la SEP considera ingreso de niños de 4 años a 2do de preescolar y siendo 1ro de preescolar no obligatorio (Jalisco, 2022), para esta investigación asumirermos que en su mayoría, ningún niño con edad menos a 3 años asiste a la escuela. <ins>Decidimos reemplazar los nulos de asistencia a la escuela (considerando preescolar como la escolaridad más baja)</ins>.

> Fuente: 
Jalisco. (2022). EN FEBRERO SON LAS PREINSCRIPCIONES A EDUCACIÓN BÁSICA PARA EL CICLO ESCOLAR 2022-2023. Recuperado de: https://www.jalisco.gob.mx/es/gobierno/comunicados/en-febrero-son-las-preinscripciones-educacion-basica-para-el-ciclo-escolar-2022

#### Hablantes de lengua indígena

In [388]:
df_limpieza_ed['hli'].value_counts(dropna=False,normalize=True)
df_limpieza_ed.loc[df_limpieza_ed['hli']==1, ['edad','ed_niv', 'ed_inas','hli']]

0.0    0.959175
1.0    0.040825
Name: hli, dtype: float64

edad  ed_niv  ed_inas  hli
12343      3     0.0      0.0  1.0
19059      3     0.0      1.0  1.0
19064      3     0.0      1.0  1.0
38740      3     0.0      0.0  1.0
63262      3     0.0      0.0  1.0
...      ...     ...      ...  ...
297886     3     0.0      0.0  1.0
299406     3     0.0      0.0  1.0
299427     3     0.0      1.0  1.0
299548     3     0.0      0.0  1.0
300283     3     0.0      0.0  1.0

[188 rows x 4 columns]

A partir de los valores de habla de lengua indígena para niños de 3 años, observamos que un 95.92% no lo habla, mientras que un 4.08% sí habla una lengua indígena. Similarmente al nivel educativo, vemos que el caso de habla indígena es más improbable que probable. Además, se debería considerar que los niños menores de 3 años tendrían que saber hablar y vivir/estar en contacto directo con un cuidador que también hable alguna lengua o vivir en una población indígena. Debido a las probabilidades observadas y considerando la edad de los encuestados, <ins>cecidimos reemplazar los nulos de habla indígena con un 0</ins>.  

Otra de las posibles maneras de abordar estos datos, podría ser la creación de un modelo predictivo en base a probabilidad de habla indígena según edad, además de considerar el área en donde el menor encuestado vive.

#### Reemplazo de Nulos

In [389]:
df.loc[df['edad'] < 3, ['ed_niv', 'ed_inas','hli']] = 0,0,0

### Limpieza de columna discap !!!! TODO !!!!

A partir del análisis del habla indígena, la inasistencia y el nivel educativo, creemos que la edad puede estar fuertemente relacionada a 

In [390]:
df_limpieza_discap = df[df['discap'].isna()][["edad","discap","pea"]]    
df_limpieza_discap

edad  discap  pea
848       49     NaN  0.0
868       60     NaN  0.0
1023      50     NaN  0.0
1383      36     NaN  0.0
1417      36     NaN  0.0
...      ...     ...  ...
307616    74     NaN  0.0
307986    84     NaN  0.0
308822    86     NaN  0.0
309221    83     NaN  0.0
309434    78     NaN  0.0

[400 rows x 3 columns]

In [391]:
df_limpieza_discap['pea'].value_counts(dropna=False,normalize=True)

0.0    0.9925
2.0    0.0075
Name: pea, dtype: float64

In [392]:
df[df['discap']==1][["edad","discap","pea"]]    

edad  discap  pea
8         87     1.0  0.0
13        57     1.0  1.0
25        69     1.0  1.0
26        70     1.0  0.0
52        68     1.0  0.0
...      ...     ...  ...
309433    87     1.0  0.0
309463    76     1.0  1.0
309466    33     1.0  2.0
309488    68     1.0  0.0
309500    84     1.0  0.0

[21846 rows x 3 columns]

### Limpieza de otros indicadores

Las columnas con valores restantes, son indicadores. Para idear maneras de limpieza a estos datos, primero hay que tener en cuenta de que hay dependencias entre ellos, ya que algunos son calculados sobre otros. Entre paréntesis se indican la cantidad de nulos por columna.

**Dependen de ic**
+ ic_1(14) y ic_3(14): son calculadas a partir del conteo de indicadores de carencias que tiene la persona.
+ vul_ic(14) y int_vul_ic(14): vulnerabilidad calculada a partir de carencias.
    + **Dependen de vulnerabilidad**: no_pob_vul(14), vul_ing(14), cuadrantes(14)
+ int_ic(14): intensidad calculada a partir de carencias.
+ pobreza(14), pobreza_m(14) y pobreza_e(14):
    + **De pobreza dependen**: int_pob_e(14), int_pob(14), priv_prof(14), priv(14)

IC incluye las carencias de servicios básicos (icsb) y de carencias de vivienda (icv). Ambos tienen 7 nulos faltantes, y entre algunas variables de las que dependen, se encuentran icsb_comb y icv_pisos respectivamente, que también tienen 7 variables faltantes. Primero analizamos estas 4 variables, para verificar que los nulos son dados gracias a los datos faltantes de icsb_comb y icsv_pisos.

In [393]:
df_limpieza_ind = df[df['icsb'].isna()]
df_limpieza_ind = df_limpieza_ind.append(df[df['icv'].isna()])
df_limpieza_ind = df_limpieza_ind[["folioviv","foliohog","numren","icsb","icsb_comb","icv","icv_pisos","ic_1","ic_3","vul_ic","int_vul_ic","vul_ing","no_pob_vul","cuadrantes","int_ic","pobreza_m","pobreza_e","int_pob_e","pobreza","int_pob","priv","priv_prof",]]
df_limpieza_ind
df_limpieza_ind.isnull().sum()

folioviv  foliohog  numren  icsb  icsb_comb  icv  icv_pisos  ic_1  \
21356    260304608         1       1   NaN        NaN  1.0        0.0   NaN   
54384    600372506         1       1   NaN        NaN  0.0        0.0   NaN   
167693  1760142722         1       1   NaN        NaN  1.0        0.0   NaN   
167694  1760142722         1       2   NaN        NaN  1.0        0.0   NaN   
167695  1760142722         1       3   NaN        NaN  1.0        0.0   NaN   
167696  1760142722         1       4   NaN        NaN  1.0        0.0   NaN   
222439  2301910005         1       1   NaN        NaN  0.0        0.0   NaN   
101656  1060357601         1       1   0.0        0.0  NaN        NaN   NaN   
101657  1060357601         1       2   0.0        0.0  NaN        NaN   NaN   
101658  1060357601         1       3   0.0        0.0  NaN        NaN   NaN   
101659  1060357601         1       4   0.0        0.0  NaN        NaN   NaN   
101660  1060357601         1       5   0.0        0.0  NaN        NaN   NaN   
101661  1060357601         1       6   0.0        0.0  NaN        NaN   NaN   
101662  1060357601         1       7   0.0        0.0  NaN        NaN   NaN   

        ic_3  vul_ic  ...  no_pob_vul  cuadrantes  int_ic  pobreza_m  \
21356    NaN     NaN  ...         NaN         NaN     NaN        NaN   
54384    NaN     NaN  ...         NaN         NaN     NaN        NaN   
167693   NaN     NaN  ...         NaN         NaN     NaN        NaN   
167694   NaN     NaN  ...         NaN         NaN     NaN        NaN   
167695   NaN     NaN  ...         NaN         NaN     NaN        NaN   
167696   NaN     NaN  ...         NaN         NaN     NaN        NaN   
222439   NaN     NaN  ...         NaN         NaN     NaN        NaN   
101656   NaN     NaN  ...         NaN         NaN     NaN        NaN   
101657   NaN     NaN  ...         NaN         NaN     NaN        NaN   
101658   NaN     NaN  ...         NaN         NaN     NaN        NaN   
101659   NaN     NaN  ...         NaN         NaN     NaN        NaN   
101660   NaN     NaN  ...         NaN         NaN     NaN        NaN   
101661   NaN     NaN  ...         NaN         NaN     NaN        NaN   
101662   NaN     NaN  ...         NaN         NaN     NaN        NaN   

        pobreza_e  int_pob_e  pobreza  int_pob  priv  priv_prof  
21356         NaN        NaN      NaN      NaN   NaN        NaN  
54384         NaN        NaN      NaN      NaN   NaN        NaN  
167693        NaN        NaN      NaN      NaN   NaN        NaN  
167694        NaN        NaN      NaN      NaN   NaN        NaN  
167695        NaN        NaN      NaN      NaN   NaN        NaN  
167696        NaN        NaN      NaN      NaN   NaN        NaN  
222439        NaN        NaN      NaN      NaN   NaN        NaN  
101656        NaN        NaN      NaN      NaN   NaN        NaN  
101657        NaN        NaN      NaN      NaN   NaN        NaN  
101658        NaN        NaN      NaN      NaN   NaN        NaN  
101659        NaN        NaN      NaN      NaN   NaN        NaN  
101660        NaN        NaN      NaN      NaN   NaN        NaN  
101661        NaN        NaN      NaN      NaN   NaN        NaN  
101662        NaN        NaN      NaN      NaN   NaN        NaN  

[14 rows x 22 columns]

folioviv       0
foliohog       0
numren         0
icsb           7
icsb_comb      7
icv            7
icv_pisos      7
ic_1          14
ic_3          14
vul_ic        14
int_vul_ic    14
vul_ing       14
no_pob_vul    14
cuadrantes    14
int_ic        14
pobreza_m     14
pobreza_e     14
int_pob_e     14
pobreza       14
int_pob       14
priv          14
priv_prof     14
dtype: int64

A partir de esta búsqueda, encontramos que <ins>los causantes de las 14 observaciones nulas en las columnas, son los 7 datos faltantes en icsb_comb y los otros 7 datos faltantes en icv_pisos</ins>.  
Además, podemos ver que la información faltante afecta a la misma casa (llave compuesta de folioviv + foliohog). Estos datos faltantes no se pueden asumir, por lo que una idea es corroborar si hay información faltante en todos los miembros encuestados de esa casa. 
* En caso de encontrar otro miembro de la casa, se podrían llenar los mismos datos de acceso a combustible y total de pisos en la vivienda.

In [394]:
# Encontramos la cantidad de residentes en cada vivienda, a partir de los datos de los 14 nulos
df_limpieza_ind[['folioviv','foliohog','numren']].groupby(['folioviv','foliohog']).count()

# Lista de IDs encontrados
ids_limpieza_ind = df_limpieza_ind['folioviv'].tolist()

# Encontramos la cantidad de residentes en cada vivienda, a partir del dataset original
df_limpieza_ind_2 = df.loc[df['folioviv'].isin(ids_limpieza_ind)][['folioviv','foliohog','numren']]
df_limpieza_ind_2.groupby(['folioviv','foliohog']).count()

numren
folioviv   foliohog        
260304608  1              1
600372506  1              1
1060357601 1              7
1760142722 1              4
2301910005 1              1

numren
folioviv   foliohog        
260304608  1              1
600372506  1              1
1060357601 1              7
1760142722 1              4
2301910005 1              1

Como se pudo ver, no hay ningún residente extra en la casa de quienes tuvieron un valor nulo. Por lo tanto, no se puede obtener información verídica sobre el acceso a combustible y los pisos de la vivienda. Por ello y dado que son sólo 14 observaciones de más de 350k, <ins>se decidió eliminar estas observaciones del dataset</ins>.

In [395]:
df = df[~df['folioviv'].isin(ids_limpieza_ind)]

### Verificaciones Finales

In [396]:
# Verificamos que ya no hay datos faltantes
null_data = df.isnull().sum()
null_data = null_data[null_data > 0].sort_values(ascending=False)
null_data

discap    400
dtype: int64

In [397]:
# Podemos observar que no hay duplicaciones de datos
df.duplicated().sum()

0

## Exportación  
Es un df.to_csv()

## Análisis Univariado #1
Michelle Figueroa Jacques is727328

In [21]:
nominales_1 = []
ordinales_1 = [] 
intervalo_1 = []
razon_1 = []

In [22]:
# Conversión de nominales a categóricos
for i in nominales_1:
    df[i] = df[i].astype('category')
    

### Cardinalidad y Exploración de Muestras
Estos se realizan para Categóricas Nominales y Ordinales

#### Cardinalidad

In [23]:
for i in nominales_1 + ordinales_1:
    print(i)
    df[i].unique()
    print()    

##### Hallazgo

#### Conteo

In [24]:
for i in nominales_1 + ordinales_1:
    print(i)
    df[i].value_counts()
    print()

##### Hallazgo

#### Transformaciones Adicionales

### Estadística Descriptiva


In [26]:
for i in nominales_1 + ordinales_1:
    print(i)
    df[i].describe()
    print()

### Gráficos


## Análisis Univariado #2
Sofía Vargas Aceves is727375

In [ ]:
nominales_2 = []
ordinales_2 = [] 
intervalo_2 = []
razon_2 = []

In [ ]:
# Conversión de nominales a categóricos
for i in nominales_2:
    df[i] = df[i].astype('category')
    

### Cardinalidad y Exploración de Muestras
Estos se realizan para Categóricas Nominales y Ordinales

#### Cardinalidad

In [ ]:
for i in nominales_2 + ordinales_2:
    print(i)
    df[i].unique()
    print()    

##### Hallazgo

#### Conteo

In [ ]:
for i in nominales_2 + ordinales_2:
    print(i)
    df[i].value_counts()
    print()

##### Hallazgo

#### Transformaciones Adicionales

### Estadística Descriptiva


In [ ]:
for i in nominales_2 + ordinales_2:
    print(i)
    df[i].describe()
    print()

### Gráficos


## Análisis Univariado #3
Gabriel

In [ ]:
nominales_3 = []
ordinales_3 = [] 
intervalo_3 = []
razon_3 = []

In [ ]:
# Conversión de nominales a categóricos
for i in nominales_3:
    df[i] = df[i].astype('category')
    

### Cardinalidad y Exploración de Muestras
Estos se realizan para Categóricas Nominales y Ordinales

#### Cardinalidad

In [ ]:
for i in nominales_3 + ordinales_3:
    print(i)
    df[i].unique()
    print()    

##### Hallazgo

#### Conteo

In [ ]:
for i in nominales_3 + ordinales_3:
    print(i)
    df[i].value_counts()
    print()

##### Hallazgo

#### Transformaciones Adicionales

### Estadística Descriptiva


In [ ]:
for i in nominales_3 + ordinales_3:
    print(i)
    df[i].describe()
    print()

### Gráficos


## Análisis Univariado #4
Miguel Mario Méndez

In [ ]:
nominales_4 = []
ordinales_4 = [] 
intervalo_4 = []
razon_4 = []

In [ ]:
# Conversión de nominales a categóricos
for i in nominales_4:
    df[i] = df[i].astype('category')
    

### Cardinalidad y Exploración de Muestras
Estos se realizan para Categóricas Nominales y Ordinales

#### Cardinalidad

In [ ]:
for i in nominales_4 + ordinales_4:
    print(i)
    df[i].unique()
    print()    

##### Hallazgo

#### Conteo

In [ ]:
for i in nominales_4 + ordinales_4:
    print(i)
    df[i].value_counts()
    print()

##### Hallazgo

#### Transformaciones Adicionales

### Estadística Descriptiva


In [ ]:
for i in nominales_4 + ordinales_4:
    print(i)
    df[i].describe()
    print()

### Gráficos
